posthog/warehouse/models/util.py

```diff
@@ -21,9 +21,17 @@
 def get_view_or_table_by_name(team, name) -> Union["DataWarehouseSavedQuery", "DataWarehouseTable", None]:
     from posthog.warehouse.models import DataWarehouseSavedQuery, DataWarehouseTable
 
+    table_name = name
+    if "." in name:
+        chain = name.split(".")
+        if len(chain) == 2:
+            table_name = f"{chain[0]}_{chain[1]}"
+        elif len(chain) == 3:
+            table_name = f"{chain[1]}_{chain[0]}_{chain[2]}"
```

Comment:
I really dont know why this would be the case, but I'll trust you've tested it. My revenue stuff is completely broken because of all these changes. I've attempted to fix it in [`c303615` (#31227)](https://github.com/PostHog/posthog/pull/31227/commits/c3036159b1aa9dc1d357e5da26005604bf3d7c35) but I'm sure this will break it again, fun stuff

Let me know once this is 100% fixed, then I can go back and re-fix mine only once

Reply:
> I really dont know why this would be the case

As in, why the ordering of the chain? The old table names are `prefix_sourceType_tableName` the new dot notation names are `sourceType.prefix.tableName`

Reply:
Ah, of course, that makes sense. The problem is that my views don't ever have the underscore, so this will break them again.

Anyway, will wait for your fix, I'm working on a branch that hasn't been merged on master yet, so it's working just fine now

Reply:
Oh this only affects DataWarehouseTables - the views work as normal here